allennlp/models/encoder_decoders/nlvr_semantic_parser.py

```diff
@@ -225,17 +228,18 @@ def forward(self,  # type: ignore
             self._agenda_coverage(in_agenda_ratio)
         return outputs
 
-    def _get_checklist_target(self,
-                              agenda: torch.LongTensor,
-                              all_actions: List[ProductionRuleArray]) -> Tuple[torch.LongTensor,
-                                                                               torch.LongTensor]:
+    def _get_checklist_info(self,
+                            agenda: torch.LongTensor,
+                            all_actions: List[ProductionRuleArray]) -> Tuple[torch.Tensor,
+                                                                             torch.Tensor,
+                                                                             torch.Tensor]:
         """
         Takes an agenda and a list of all actions and returns a target checklist against which the
-        checklist at each state will be compared against to compute a loss, and the indices of
-        actions relevant for checklist loss computation (``relevant_actions``). If
-        ``penalize_non_agenda_actions`` is set to ``True``, ``relevant_actions`` will contain
-        indices to all the terminal actions. If not, we will simply return the ``agenda`` itself as
-        ``relevant_actions``, in which case, it will be a padded tensor, with -1 indicating padding.
+        checklist at each state will be compared to compute a loss, indices of ``terminal_actions``,
+        and a ``checklist_mask`` that indicates which of the terminal actions are relevant for
+        checklist loss computation. If ``penalize_non_agenda_actions`` is set to``True``,
```

Comment:
I think you want `self._penalize_non_agenda_actions` here - this statement makes it look like this is an argument to the function, not some internal state.